plaso/analysis/interface.py

```diff
@@ -198,37 +199,38 @@ def _EnsureRequesterStarted(self):
       self._analyzer.start()
       self._analyzer_started = True
 
-  def ExamineEvent(self, mediator, event):
+  def ExamineEvent(self, mediator, event, event_data):
     """Evaluates whether an event contains the right data for a hash lookup.
 
     Args:
       mediator (AnalysisMediator): mediates interactions between
           analysis plugins and other components, such as storage and dfvfs.
       event (EventObject): event.
+      event_data (EventData): event data.
     """
+    if (event_data.data_type not in self.DATA_TYPES or
+        not self._analyzer.lookup_hash):
+      return
+
     self._EnsureRequesterStarted()
 
-    path_spec = event.pathspec
-    event_identifiers = self._event_identifiers_by_pathspec[path_spec]
+    event_identifiers = self._event_identifiers_by_pathspec[event_data.pathspec]
```

Comment:
Pathspec is set on the event, not the data: https://github.com/log2timeline/plaso/blob/master/plaso/parsers/mediator.py#L423

Reply:
Per above, it looks like the pathspec is set on EventData, but it's not defined there: https://github.com/log2timeline/plaso/blob/master/plaso/containers/events.py

Can you please clarify what's intended here? It would also be helpful to have a clear statement about what sort of data belongs in each other. If I have the same file, copy it to a different machine and rename it, should the eventdatas produced from the files be the same? Can both the events and eventdatas be different?

Reply:
> Pathspec is set on the event, not the data: 

It is set on event data at https://github.com/log2timeline/plaso/blob/master/plaso/parsers/mediator.py#L423

This might be a bit hard to follow (and needs cleanup, hence this PR)

But it is passed event data from https://github.com/log2timeline/plaso/blob/master/plaso/parsers/mediator.py#L503

> Can you please clarify what's intended here?

it should be set on event data, event should only have a restrictive set of attributes event should be something like:
(timestamp, timestamp_desc, event_data)